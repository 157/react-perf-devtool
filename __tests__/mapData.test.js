const generateDataFromMeasures = require('../src/utils/generate')
const getReactPerformanceData = require('../src/utils/parse')
const MEASURES = require('../samples/measures')

describe("Map Data from React Performance Data", () => {
  test("Returns an object containing information about the component measure", () => {
    console.log(generateDataFromMeasures(getReactPerformanceData(MEASURES)))
    expect(generateDataFromMeasures(getReactPerformanceData(MEASURES))).toMatchSnapshot()
  })
})
