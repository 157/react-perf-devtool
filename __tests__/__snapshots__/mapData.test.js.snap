// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Data from React Performance Data Returns an object containing information about the component measure 1`] = `
Array [
  Object {
    "componentDidMount": Object {
      "averageTimeSpentMs": "0.92",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.92,
    },
    "componentDidUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentName": "App",
    "componentWillMount": Object {
      "averageTimeSpentMs": "3.02",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 3.02,
    },
    "componentWillReceiveProps": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUnmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "mount": Object {
      "averageTimeSpentMs": "36.88",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 36.88,
    },
    "numberOfInstances": 1,
    "percentTimeSpent": "35%",
    "render": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "shouldComponentUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "totalTimeSpent": 40.82000000000001,
    "unmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "update": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
  },
  Object {
    "componentDidMount": Object {
      "averageTimeSpentMs": "0.81",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.81,
    },
    "componentDidUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentName": "Main",
    "componentWillMount": Object {
      "averageTimeSpentMs": "0.88",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.88,
    },
    "componentWillReceiveProps": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUnmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "mount": Object {
      "averageTimeSpentMs": "31.35",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 31.35,
    },
    "numberOfInstances": 1,
    "percentTimeSpent": "29%",
    "render": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "shouldComponentUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "totalTimeSpent": 33.040000000000006,
    "unmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "update": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
  },
  Object {
    "componentDidMount": Object {
      "averageTimeSpentMs": "0.46",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.46,
    },
    "componentDidUpdate": Object {
      "averageTimeSpentMs": "0.81",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.81,
    },
    "componentName": "Introduction",
    "componentWillMount": Object {
      "averageTimeSpentMs": "0.84",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.84,
    },
    "componentWillReceiveProps": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUnmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "mount": Object {
      "averageTimeSpentMs": "11.33",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 11.33,
    },
    "numberOfInstances": 1,
    "percentTimeSpent": "14%",
    "render": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "shouldComponentUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "totalTimeSpent": 16.27,
    "unmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "update": Object {
      "averageTimeSpentMs": "2.83",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 2.83,
    },
  },
  Object {
    "componentDidMount": Object {
      "averageTimeSpentMs": "0.08",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.08,
    },
    "componentDidUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentName": "Projects",
    "componentWillMount": Object {
      "averageTimeSpentMs": "1.85",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 1.85,
    },
    "componentWillReceiveProps": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUnmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "mount": Object {
      "averageTimeSpentMs": "9.44",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 9.44,
    },
    "numberOfInstances": 1,
    "percentTimeSpent": "13%",
    "render": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "shouldComponentUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "totalTimeSpent": 14.399999999999999,
    "unmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "update": Object {
      "averageTimeSpentMs": "3.03",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 3.03,
    },
  },
  Object {
    "componentDidMount": Object {
      "averageTimeSpentMs": "0.83",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.83,
    },
    "componentDidUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentName": "Posts",
    "componentWillMount": Object {
      "averageTimeSpentMs": "1.11",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 1.11,
    },
    "componentWillReceiveProps": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUnmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "mount": Object {
      "averageTimeSpentMs": "4.05",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 4.05,
    },
    "numberOfInstances": 1,
    "percentTimeSpent": "5%",
    "render": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "shouldComponentUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "totalTimeSpent": 5.99,
    "unmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "update": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
  },
  Object {
    "componentDidMount": Object {
      "averageTimeSpentMs": "0.55",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.55,
    },
    "componentDidUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentName": "Message",
    "componentWillMount": Object {
      "averageTimeSpentMs": "0.67",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 0.67,
    },
    "componentWillReceiveProps": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUnmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "componentWillUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "mount": Object {
      "averageTimeSpentMs": "3.41",
      "numberOfTimes": 1,
      "totalTimeSpentMs": 3.41,
    },
    "numberOfInstances": 1,
    "percentTimeSpent": "4%",
    "render": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "shouldComponentUpdate": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "totalTimeSpent": 4.63,
    "unmount": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
    "update": Object {
      "averageTimeSpentMs": "-",
      "numberOfTimes": 0,
      "totalTimeSpentMs": 0,
    },
  },
]
`;
